recipes/functions-framework-cpp/all/conanfile.py

```diff
@@ -0,0 +1,134 @@
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.33.0"
+
+
+class FunctionsFrameworkCppConan(ConanFile):
+    name = "functions-framework-cpp"
+    description = "An open source FaaS (Functions as a Service) framework"
+    license = "Apache-2.0"
+    topics = "google", "cloud", "functions-as-a-service", "faas-framework"
+    homepage = "https://github.com/GoogleCloudPlatform/functions-framework-cpp"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package_multi", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    short_paths = True
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "9",
+            "Visual Studio": "15.7",
+            "clang": "7",
+            "apple-clang": "11",
+        }
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 17)
+        minimum_version = self._compilers_minimum_version.get(
+            str(self.settings.compiler), False
+        )
+        if minimum_version:
+            if tools.Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    "{} requires C++17, which your compiler does not support.".format(
+                        self.name
+                    )
+                )
+        else:
+            self.output.warn(
+                "{} requires C++17. Your compiler is unknown. Assuming it supports C++17.".format(
+                    self.name
+                )
+            )
+
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("Fails to compile for Windows as a DLL")
+
+        if hasattr(self, "settings_build") and tools.cross_building(self):
+            raise ConanInvalidConfiguration(
+                "Recipe not prepared for cross-building (yet)"
+            )
+
+    def source(self):
+        tools.get(
+            **self.conan_data["sources"][self.version],
+            destination=self._source_subfolder,
+            strip_root=True
+        )
+
+    def requirements(self):
+        self.requires("abseil/20210324.2")
+        self.requires("boost/1.75.0")
```

Comment:
```suggestion
        self.requires("boost/1.77.0")
```

Latest version available: https://conan.io/center/boost

Reply:
Done (and rebased)